### PR TITLE
initial code review for job alerts

### PR DESCRIPTION
**Pull request by sureshpegadapelli**

### CODE REVIEW

File: application/models/UserCategoryMapModel.php [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+             * after getting the user details send alerts
+             */
+            foreach ($user_detail_result as $user_details) {
+                
+                /*$this->load->library('email');
+                $this->email->from('onesearch@gmail.com', 'One Search');
+                $this->email->to($user_details->email);
+                $this->email->subject('Job Alerts');
+                $this->email->message($jobtitle);
+                if (!$this->email->send()) {
```
Comment (abhinaykrupa) [inline application/models/UserCategoryMapModel.php]: to send out emails create a completely new controller which specifically deals with sending out emails to customers ... this approach will help us use some external emailing services like amzon going forward and we dont have to change it all over the place instead just change it in the email controller

File: application/models/UserCategoryMapModel.php [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                $this->db->where($where);
+                $this->db->update('user_category_map', $category_map_array);
+                echo $job_details;
+                
+            }
+            
+        }
+    }
+
+    function getUsersandMetavalues($cat_id, $apiurl, $apikey) {
```
Comment (abhinaykrupa) [inline application/models/UserCategoryMapModel.php]: A in and 

V in values 

has to be capital letters

**Comment by sureshpegadapelli** [comment]
Testing the pull request

File: application/models/UserCsvModel.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
+<?php
```
Comment (abhinaykrupa) [inline application/models/UserCsvModel.php]: rename this file to importCSVToModel.php

File: application/views/uploadCsvView.php [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<form action="<?php echo site_url();?>Userdatauploadcsv/uploadData" method="post" enctype="multipart/form-data" name="form1" id="form1">
```
Comment (abhinaykrupa) [inline application/views/uploadCsvView.php]: rename this to uploadCSVView.php and add a drop down based on what your are uploading

Comment (sureshpegadapelli) [reply application/views/uploadCsvView.php]: Down

Comment (sureshpegadapelli) [reply application/views/uploadCsvView.php]: Done

File: application/models/UserCsvModel.php (hunk 3 above)
Comment (sureshpegadapelli) [reply application/models/UserCsvModel.php]: Done

File: application/models/UserCategoryMapModel.php (hunk 2 above)
Comment (sureshpegadapelli) [reply application/models/UserCategoryMapModel.php]: Done

File: application/libraries/Curl.php [hunk 5]
```diff
@@ -0,0 +1,327 @@
+<?php
```
[the remark this answers is not in the document]
Comment (sureshpegadapelli) [reply application/libraries/Curl.php]: File is removed

File: application/models/UserCategoryMapModel.php [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+    function __construct() {
+        parent::__construct();
+    }
+
+    public function getUsersMetavalues($cat_id, $api_url, $api_key) {
+        /**
+         * Query for getting the meta pair with count
+         */
+        $meta_query = $this->db->query("select
```
[the remark this answers is not in the document]
Comment (sureshpegadapelli) [reply application/models/UserCategoryMapModel.php]: Done

File: application/models/UserCategoryMapModel.php [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    }
+
+    public function getUsersMetavalues($cat_id, $api_url, $api_key) {
+        /**
+         * Query for getting the meta pair with count
+         */
+        $meta_query = $this->db->query("select 
+        count(*) as count,search,location
+        from 
+        (select
```
[the remark this answers is not in the document]
Comment (sureshpegadapelli) [reply application/models/UserCategoryMapModel.php]: Done

File: application/controllers/UserCategoryMap.php [hunk 8]
```diff
@@ -0,0 +1,72 @@
+<?php
+defined('BASEPATH') OR exit('No direct script access allowed');
+
+class UserCategoryMap extends MY_Controller
+{
+    var $search;
+    var $location;
+    var $radius_miles;
```
[the remark this answers is not in the document]
Comment (abhinaykrupa) [reply application/controllers/UserCategoryMap.php]: @sureshpegadapelli  how about this

File: application/models/UserCategoryMapModel.php (hunk 1 above)
Comment (abhinaykrupa) [reply application/models/UserCategoryMapModel.php]: how about this? @sureshpegadapelli

**Pull request merged**